pkg/cloud/services/ec2/instances.go

```diff
@@ -420,13 +420,16 @@ func (s *Service) GetCoreNodeSecurityGroups(scope *scope.MachinePoolScope) ([]st
 	}
 
 	ids := make([]string, 0, len(sgRoles))
-	for _, sg := range sgRoles {
-		if _, ok := s.scope.SecurityGroups()[sg]; !ok {
-			return nil, awserrors.NewFailedDependency(
-				fmt.Sprintf("%s security group not available", sg),
-			)
+	existingSGs := s.scope.SecurityGroups()
+	if len(existingSGs) != 0 {
```

Comment:
Generally we follow convention len(<variable>) > 0

Reply:
is this addressed?

Reply:
done